apps/vaporgui/VLineEdit.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include <iostream>
+#include <string>
+
+#include "VLineEdit.h"
+
+VLineEdit::VLineEdit( const std::string& value )
+: VContainer( this ),
+  _value( value ),
+  _isDouble( false )
+{
+    _lineEdit = new QLineEdit;
+    SetValue( _value );
+    layout()->addWidget(_lineEdit);
+
+    connect( _lineEdit, SIGNAL( editingFinished() ),
+        this, SLOT( emitLineEditChanged() ) );
+}
+
+void VLineEdit::SetValue( const std::string& value ) {
+    // see if value is a valid double
+    if (_isDouble) {
+        try {
+            double dValue = std::stod( value );
+            _value = value;
+        }
+        catch (...) {
+            std::cerr << "VLineEDit::SetValue failed to set value of " << value << std::endl;
```

Comment:
Shouldn't be sending diagnostics to stderr. Either use our GUI error reporting mechanism, or reject the input.

Reply:
Fixed